webserver/src/main/java/io/kestra/webserver/controllers/ExecutionController.java

```diff
@@ -736,14 +738,33 @@ public static class StateRequest {
     @Operation(tags = {"Executions"}, summary = "Kill an execution")
     @ApiResponse(responseCode = "204", description = "On success")
     @ApiResponse(responseCode = "409", description = "if the executions is already finished")
+    @ApiResponse(responseCode = "404", description = "if the executions is not found")
     public HttpResponse<?> kill(
         @Parameter(description = "The execution id") @PathVariable String executionId
-    ) {
-        Optional<Execution> execution = executionRepository.findById(executionId);
-        if (execution.isPresent() && execution.get().getState().isTerminated()) {
+    ) throws InternalException {
+        Optional<Execution> maybeExecution = executionRepository.findById(executionId);
+        if (maybeExecution.isEmpty()) {
+            return HttpResponse.notFound();
+        }
+
+        var execution = maybeExecution.get();
+        if (execution.getState().isTerminated()) {
             throw new IllegalStateException("Execution is already finished, can't kill it");
         }
 
+        if (execution.getState().isPaused()) {
```

Comment:
same modification than for the pause one: move to service, more unit, throw exception, ...